### PR TITLE
{AKS} Set default vm-set-type and ignore spaces for Mixed SKU

### PR DESCRIPTION
**Pull request by HuShaoRu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks create --vm-set-type --vm-sizes
az aks nodepool add --vm-set-type --vm-sizes
az aks nodepool manual-scale

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/agentpool_decorator.py [hunk 1]
```diff
@@ -86,7 +86,17 @@ def get_vm_set_type(self) -> str:
         :return: string
         """
         # read the original value passed by the command
-        vm_set_type = self.raw_param.get("vm_set_type", CONST_VIRTUAL_MACHINE_SCALE_SETS)
+        vm_set_type = self.raw_param.get("vm_set_type")
+        if vm_set_type is None:
+            if self.raw_param.get("vm_sizes") is None:
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/agentpool_decorator.py:91]: Is it possible to replace `self.raw_param.get("vm_sizes")` with `self.get_vm_sizes()`?

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=95198173&view=results)

- test_vms_agentpool_type

File: src/aks-preview/azext_aks_preview/agentpool_decorator.py (hunk 1 above)
Comment (HuShaoRu) [reply src/aks-preview/azext_aks_preview/agentpool_decorator.py:91]: `self.get_vm_sizes()` will return default value if the param is not given. Here we only want to check user's input.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**